examples/advanced/pi-lluminate.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+
+# Use the Enviro pHAT to check how much light is in a room, then trigger a webhook to control
+# Philips Hue lights via IFTTT. By Wesley Archer (@raspberrycoulis).
+
+# Import the relevant modules
+
+import requests
+import time
+import datetime
+import os
+from envirophat import light
+
+# Get the current time for displaying in the terminal.
+def whats_the_time():
+    now = datetime.datetime.now()
+    return (now.strftime("%H:%M:%S"))
+
+# The function to turn off the lights. Sends a webhook to IFTTT which
+# triggers Hue. Replace the trigger word and tokens from your account.
+
+def turn_off():
+    requests.post("https://maker.ifttt.com/trigger/{TRIGGER_WORD}/with/key/{TOKEN_GOES_HERE}")
+    print("Lights off!")
+
+# The function to turn on the lights. Sends a webhook to IFTTT which                   
+# triggers Hue. Replace the trigger word and tokens from your account.
+
+def turn_on():
+    requests.post("https://maker.ifttt.com/trigger/{TRIGGER_WORD}/with/key/{TOKEN_GOES_HERE}")
+    print("Lights on!")
+
+# Check the light level and determine whether the lights need to 
+# be turned on or off.
+
+def average_lux():
+    # Variables for calculating the average lux levels
+    start_time = time.time()
+    curr_time = time.time()
+    collect_light_time = 60
+    collect_light_data = []
+
+    # Calculate the average lux level over 60 seconds
+    print("Calculating average light level...")
+    while curr_time - start_time < collect_light_time:
+	      curr_time = time.time()
+	      avg = light.light()
+	      collect_light_data.append(avg)
+	  time.sleep(1)
+    average_light = sum(collect_light_data[-10:]) / 10.0
```

Comment:
It looks like you're collecting an arbitrary (well, roughly 60) number of samples here and discarding all but the last 10?

You can average over your whole set with:

```python
average_light = sum(collect_light_data) / float(len(collect_light_data))
```

If you intend to use only the last 10 samples, you should include a comment that rationalises why.

Reply:
This would be a bit I forgot to update when I submitted the PR after making it more generic from my own version (where I use averages over 15 seconds). I picked 10 as I felt this would be a good enough value from the 15 second data sample.

Happy to update the range as it should increase the accuracy too - I also tweaked the high threshold too, as some Hue lights (if set to a brighter white) can trigger the `turn_off()` function.